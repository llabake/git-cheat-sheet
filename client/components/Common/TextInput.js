import React from 'react';

const TextInput = ({
  name, onChange, placeholder, value, error, icon, type, 
}) => (
  <div className="form-field">
    <i className="material-icons icon user-icon">{icon}</i>
    <input
        type={type}
        name={name}
        className="form-control"
        placeholder={placeholder}
        value={value}
        onChange={onChange}
      />
    <div className="error">
      {error}
    </div>
  </div>
);

export default TextInput;
