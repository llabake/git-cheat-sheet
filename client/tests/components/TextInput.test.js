import React from 'react';
import { mount } from 'enzyme';

import TextInput from '../../components/Common/TextInput';

const setup = () => {
  const props = {
    placeholder: '',
    icon: '',
    name: '',
    value: '',
    onChange: jest.fn(),
  };

  const wrapper = mount(<TextInput {...props} />);

  return { wrapper, props };
};

describe('<TextInput />', () => {
  const { wrapper } = setup();

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
