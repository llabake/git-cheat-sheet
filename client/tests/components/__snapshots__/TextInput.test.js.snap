// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should render correctly 1`] = `
<TextInput
  icon=""
  name=""
  onChange={[MockFunction]}
  placeholder=""
  value=""
>
  <div
    className="form-field"
  >
    <i
      className="material-icons icon user-icon"
    />
    <input
      className="form-control"
      name=""
      onChange={[MockFunction]}
      placeholder=""
      value=""
    />
    <div
      className="error"
    />
  </div>
</TextInput>
`;
